sdk/containerregistry/container-registry/samples-dev/containerRegistryClient.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Demonstrates the use of a ContainerRegistryClient.
+ * @azsdk-weight 10
+ */
+
+import { ContainerRegistryClient } from "@azure/container-registry";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+export async function main() {
+  // endpoint should be in the form of "https://myregistryname.azurecr.io"
+  // where "myregistryname" is the actual name of your registry
+  const endpoint = process.env.CONTAINER_REGISTRY_ENDPOINT || "<endpoint>";
+  const client = new ContainerRegistryClient(endpoint, new DefaultAzureCredential());
+  await listRepositories(client);
+  await deleteRepository(client);
+}
+
+async function listRepositories(client: ContainerRegistryClient) {
+  console.log("Listing repositories");
+  const iterator = client.listRepositories();
+  for await (const repository of iterator) {
+    console.log(`  repository: ${repository}`);
+  }
+
+  console.log("  by pages");
```

Comment:
In .NET, iterating through the collection by pages -- rather than one element at a time and letting the client handle retrieving subsequent pages -- is a relatively advanced scenario and not something we distract customers with in our "getting started" documentation.  I don't mind providing a sample for it, but I would make sure the sample indicated that this is relatively advanced, and not the "getting started" path.

Reply:
That's fair. I will extract it into a new method and add some comments

Reply:
We do include paged iteration in some form or another in most package samples for packages that support it. For Form, we show how to do paged iteration of models. There, it is in a specific sample that we use to show methods of iterating over the models.

Reply:
Yes, in storage we also have separate samples iterators (because we have so many ways to use them).  I will consider doing similar in next release.

Reply:
Tracked in #14674